### PR TITLE
[feature](backup-restore) Add local backup/restore not upload/download by broker

### PR DESCRIPTION
**Pull request by JackDrogon**
## Proposed changes

Support keep snapshot on the be, not upload by borker and restore into another cluster by http 

1. Backup
```
BACKUP SNAPSHOT ccr.snapshot_20230605 TO `__keep_on_local__` ON (src_1) PROPERTIES ("type" = "full");
```
2. Get Snapshot
Get the snapshot by getSnapshot thrift rpc, fe will return snapshot meta && jobinfo

3. Restore
Rewrite jobinfo by adding extra info(master token && be addresses), use restore snapshot thrift rpc

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by JackDrogon** [comment]
run buildall

File: be/src/runtime/snapshot_loader.cpp [hunk 1]
```diff
@@ -370,6 +373,272 @@ Status SnapshotLoader::download(const std::map<std::string, std::string>& src_to
...
+            LOG(WARNING) << ss.str();
+            return Status::InternalError(ss.str());
+        }
+        DataDir* data_dir = tablet->data_dir();
+
+        // download all need download files
+        uint64_t total_file_size = 0;
+        MonotonicStopWatch watch;
+        watch.start();
+        for (auto& filename : need_download_files) {
```
Comment (dataroaring) [inline be/src/runtime/snapshot_loader.cpp:541]: We should control parallel of download via a config or a queue.

**Review by dataroaring — CHANGES_REQUESTED** [review]
Please add some regression test.

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**